x-pack/plugins/observability_solution/observability_onboarding/server/routes/flow/route.ts

```diff
@@ -171,8 +182,240 @@ const getProgressRoute = createObservabilityOnboardingServerRoute({
+  options: { tags: [], xsrfRequired: false },
+  params: t.type({
+    path: t.type({
+      onboardingId: t.string,
+    }),
+    body: t.string,
+  }),
+  async handler({ context, request, response, params, core, plugins, services }) {
+    const coreStart = await core.start();
+    const fleetStart = await plugins.fleet.start();
+    const savedObjectsClient = coreStart.savedObjects.createInternalRepository();
+    const packageClient = fleetStart.packageService.asScoped(request);
+
+    const savedObservabilityOnboardingState = await getObservabilityOnboardingFlow({
+      savedObjectsClient,
+      savedObjectId: params.path.onboardingId,
+    });
+    if (!savedObservabilityOnboardingState) {
+      throw Boom.notFound(`Onboarding session '${params.path.onboardingId}' not found.`);
+    }
+
+    const integrationsToInstall = parseIntegrationsTSV(params.body);
+    if (!integrationsToInstall.length) {
+      return response.badRequest({
+        body: {
+          message: 'Please specify a list of integrations to install',
+        },
+      });
+    }
+
+    await saveObservabilityOnboardingFlow({
+      savedObjectsClient,
+      savedObjectId: params.path.onboardingId,
+      observabilityOnboardingState: {
+        ...savedObservabilityOnboardingState,
+        type: 'logFiles',
+        progress: {},
+      } as ObservabilityOnboardingFlow,
+    });
+
+    let agentInputs: unknown[];
+    try {
+      agentInputs = await ensureInstalledIntegrations(integrationsToInstall, packageClient);
+    } catch (error) {
+      if (error instanceof FleetUnauthorizedError) {
+        return response.forbidden({
+          body: {
+            message: error.message,
+          },
+        });
+      }
+      throw error;
+    }
+
+    const authApiKey = getAuthenticationAPIKey(request);
+    const elasticsearchUrl = plugins.cloud?.setup?.elasticsearchUrl
+      ? [plugins.cloud?.setup?.elasticsearchUrl]
+      : await getFallbackESUrl(services.esLegacyConfigService);
+
+    return response.ok({
+      headers: {
+        'content-type': 'application/yaml',
+      },
+      body: generateAgentConfig({
+        apiKey: authApiKey ? `${authApiKey?.apiKeyId}:${authApiKey?.apiKey}` : '$API_KEY',
+        esHost: elasticsearchUrl,
+        inputs: agentInputs,
+      }),
+    });
+  },
+});
+
+type Integration =
+  | {
+      pkgName: string;
+      installSource: 'registry';
+    }
+  | {
+      pkgName: string;
+      installSource: 'custom';
+      logFilePaths: string[];
+    };
+
+async function ensureInstalledIntegrations(
+  integrationsToInstall: Integration[],
+  packageClient: PackageClient
+) {
+  const agentInputs: unknown[] = [];
+  for (const integration of integrationsToInstall) {
+    const { pkgName, installSource } = integration;
+    if (installSource === 'registry') {
+      await packageClient.ensureInstalledPackage({ pkgName });
+      agentInputs.push(...getSystemLogsDataStreams(uuidv4()));
```

Comment:
I tested with system integration which does not require any other inputs. What inputs do you have in mind?

Reply:
nginx, docker, apache

Reply:
This can also be handled via a separate PR, just pointing out that this will not return the right config for detected integrations - see https://github.com/elastic/kibana/issues/183363 :

> Extend config generation logic:
> * Get configs from fleet plugin
> * Combine into single elastic-agent.yml file

Reply:
> nginx, docker, apache

Oh sorry, yeh we need to get the logic from Fleet for this which is currently not possible so I would do that as a follow up. It's not ideal that we're currently replicating the config generation logic inside onboarding plugin

Reply:
Ideally we should pull all config from fleet as they have the source of truth. Happy to tackle it separately 👍